geonode/layers/views.py

```diff
@@ -812,40 +811,38 @@ def dataset_download(request, layername):
         # if GeoServer is not used, we redirect to the proxy download
         return HttpResponseRedirect(reverse('download', args=[dataset.id]))
 
-    download_format = request.GET.get('export_format', 'application/zip')
+    download_format = request.GET.get('export_format')
 
-    if not wps_format_is_supported(download_format, dataset.subtype):
+    if download_format and not wps_format_is_supported(download_format, dataset.subtype):
         logger.error("The format provided is not valid for the selected resource")
         return JsonResponse({"error": "The format provided is not valid for the selected resource"}, status=500)
 
     # getting default payload
     tpl = get_template("geoserver/dataset_download.xml")
     ctx = {
         "alternate": dataset.alternate,
-        "download_format": download_format
+        "download_format": 'application/zip' if dataset.is_vector() else 'image/tiff'
```

Comment:
@mattiagiupponi you're not using the (optional) `download_format` obtained from the `export_format` param